ttnn/cpp/ttnn/operations/data_movement/sharded_partial/interleaved_to_sharded_partial/interleaved_to_sharded_partial.cpp

```diff
@@ -0,0 +1,77 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+
+#include "ttnn/common/constants.hpp"
+#include "ttnn/run_operation.hpp"
+#include "device/interleaved_to_sharded_partial_op.hpp"
+#include "interleaved_to_sharded_partial.hpp"
+#include "ttnn/deprecated/tt_dnn/op_library/work_split.hpp"
+
+namespace ttnn::operations::data_movement{
+
+ttnn::Tensor InterleavedToShardedPartialOperation::operator()(
+    uint8_t queue_id,
+    const ttnn::Tensor& input_tensor,
+    const std::variant<CoreCoord, CoreRangeSet> & grid,
+    const std::array<uint32_t, 2> & shard_shape,
+    int64_t& num_slices,
+    int64_t& slice_index,
+    tt::tt_metal::TensorMemoryLayout shard_scheme,
+    tt::tt_metal::ShardOrientation shard_orientation,
+    const std::optional<DataType> & data_type_arg) {
+
+
+
+    std::vector<Tensor> output_tensors = {Tensor(operation::get_workers_for_op_output({input_tensor}))};
+    operation::launch_op(
```

Comment:
is launch op really needed here? its added automatically in a decorator so maybe you don't really need it